tests/test_io.py

```diff
@@ -0,0 +1,67 @@
+    example_filepath = fixture_path / "example.ply"
+    res = ply.write_ply(
+        str(example_filepath),
+        [points_array, colors_array, classification_array],
+        ["x", "y", "z", "red", "green", "blue", "classification"],
+    )
+    assert res and example_filepath.exists()
+
+    read_data = ply.read_ply(str(example_filepath))
+    read_points = np.vstack((read_data[0][:, 0], read_data[0][:, 1], read_data[0][:, 2])).T
+    read_colors = np.vstack((read_data[1][:, 0], read_data[1][:, 1], read_data[1][:, 2])).T
+    read_classification = read_data[2]
+    np.testing.assert_array_equal(points_array, read_points)
+    np.testing.assert_array_equal(colors_array, read_colors)
+    np.testing.assert_array_equal(classification_array, read_classification)
+
+
+def test_write_las(fixture_path, points_array, colors_array, classification_array):
+    example_filepath = fixture_path / "example.las"
+    res = las.write_las(str(example_filepath), points_array, colors_array, classification_array)
+    assert res and example_filepath.exists()
+
+    read_data = las.read_las_laz(str(example_filepath))
+    read_points = np.vstack((read_data[0][:, 0], read_data[0][:, 1], read_data[0][:, 2])).T
+    read_colors = np.vstack((read_data[1][:, 0], read_data[1][:, 1], read_data[1][:, 2])).T
+    read_classification = read_data[2]
+    np.testing.assert_array_equal(points_array, read_points)
+    np.testing.assert_array_equal(colors_array, read_colors)
+    np.testing.assert_array_equal(classification_array, read_classification)
+
+
+def test_write_xyz(fixture_path, points_array, colors_array, classification_array):
+    example_filepath = fixture_path / "example.xyz"
+    res = xyz.write_xyz(str(example_filepath), points_array, colors_array, classification_array)
+    assert res and example_filepath.exists()
+
+    read_data = xyz.read_xyz(str(example_filepath))
+    read_points = np.vstack((read_data[0][:, 0], read_data[0][:, 1], read_data[0][:, 2])).T
+    read_colors = np.vstack((read_data[1][:, 0], read_data[1][:, 1], read_data[1][:, 2])).T
+    read_classification = (read_data[2]).T
+    np.testing.assert_array_equal(points_array, read_points)
+    np.testing.assert_array_equal(colors_array, read_colors)
+    np.testing.assert_array_equal(classification_array, read_classification)
+
+
+def test_read_xyz(fixture_path, points_array):
+    example_filepath = fixture_path / "example.xyz"
+    points, _, _ = xyz.read_xyz(example_filepath)
+    np.testing.assert_array_equal(points_array, points)
+
+
+def test_read_ply(fixture_path, points_array):
+    example_filepath = fixture_path / "example.ply"
+    points, _, _ = ply.read_ply(example_filepath)
+    np.testing.assert_array_equal(points_array, points)
+
+
+def test_read_las(fixture_path, points_array):
+    example_filepath = fixture_path / "example.las"
+    points, _, _ = las.read_las_laz(example_filepath)
+    np.testing.assert_array_equal(points_array, points)
```

Comment:
This is a detail, but I think it is more relevant to test the colors and labels in the reader tests than in the writer tests.

The writer tests are there to check if the file is correctly written (is it on the file system? can we read the content back?), but details on the reading process should be tested in the reader tests.